navigation/waypoint_manager/scripts/waypoint_manager.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+import rclpy
+from rclpy.node import Node
+from vortex_msgs.srv import Waypoint
+from nav_msgs.msg import Path
+from geometry_msgs.msg import PoseStamped, Point
+
+
+class WaypointManager(Node):
+    """
+    Nodes created:
+        WaypointManager
+    Subscribes to:
+    Publishes to:
+    """
+    
+    def __init__(self):
+        """
+        Initializes WaypointManager class.
+
+        Initializes node, sets up action client, services, and publishers.
+        """
+        super().__init__("WaypointManager")
+
+        self.waypoint_list = []
+
+        # Service for getting the list of waypoints and client for sending the list of waypoint
+        self.waypoint_service = self.create_service(Waypoint, 'waypoint', self.waypoint_callback)
+        self.waypoint_list_client = self.create_client(Waypoint, 'waypoint_list')
+
+        # Services
+        self.get_logger().info('kommet til services')
+        self.add_waypoint_service = self.create_service(Waypoint, 'add_waypoint', self.add_waypoint_callback)
+        self.get_logger().info('kommet til etter at man har laget add_waypoint_services')
+
+        # nav_msgs Path to visualize LOS in Rviz
+        self.path_pub = self.create_publisher(Path, 'los_path', 10)
+        self.path = Path()
+        self.path.header.frame_id = 'world'
+
+    def waypoint_callback(self, request, response):
+        """
+        Callback function to handle received waypoints.
+        """
+        # Clear the existing waypoint list
+        self.waypoint_list.clear()
+
+        return self.add_waypoint_to_list(request)
+    
+   
+    def add_waypoint_to_list(self, req):
+        """
+        Adds waypoints to the waypoint list.
+
+        Args:
+            req (Waypoint.Request): Request containing the list of waypoints to be added.
+
+        Returns:
+            Waypoint.Response: True if waypoints are added successfully.
+        """
+        for i in range(0, len(req.waypoint), 2):
+            x = req.waypoint[i]
+            y = req.waypoint[i+1]
+
+            self.waypoint_list.append([x, y])
+            self.get_logger().info("Added waypoint to waypoint_list")
+
+            new_pose = PoseStamped()
+            new_pose.pose.position = Point(x=x, y=y, z=0.0)
+            self.path.poses.append(new_pose)
+            self.path_pub.publish(self.path)
+            self.get_logger().info("points :" + str(self.waypoint_list))
```

Comment:
I think this would be easier to interpret if the waypoint extraction was done beforehand, something like

```python
new_waypoints = list(zip(req.waypoint[::2], req.waypoint[1::2]))
self.waypoint_list.extend(new_waypoints)

for x, y in new_waypoints:
    new_pose = PoseStamped()
    new_pose.pose.position = Point(x=x, y=y, z=0.0)
    self.path.poses.append(new_pose)
```

(This example would need to be rewritten to work with  `Waypoint2D` objects if you end up implementing that)